tests/components/dynalite/test_init.py

```diff
@@ -1,74 +1,91 @@
 """Test Dynalite __init__."""
-from unittest.mock import Mock, call, patch
+from unittest.mock import Mock
 
-from homeassistant.components.dynalite import DATA_CONFIGS, DOMAIN, LOGGER
-from homeassistant.components.dynalite.__init__ import (
-    async_setup,
-    async_setup_entry,
-    async_unload_entry,
-)
+from asynctest import patch
 
-from tests.common import mock_coro
+from homeassistant.components import dynalite
+from homeassistant.setup import async_setup_component
 
-
-async def test_async_setup():
-    """Test a successful setup."""
-    new_host = "1.2.3.4"
-    old_host = "5.6.7.8"
-    hass = Mock()
-    hass.data = {}
-    config = {DOMAIN: {"bridges": [{"host": old_host}, {"host": new_host}]}}
-    mock_conf_host = Mock(return_value=[old_host])
-    with patch(
-        "homeassistant.components.dynalite.__init__.configured_hosts", mock_conf_host
-    ):
-        await async_setup(hass, config)
-        mock_conf_host.assert_called_once()
-        assert mock_conf_host.mock_calls[0] == call(hass)
-        assert hass.data[DOMAIN][DATA_CONFIGS] == {
-            new_host: {"host": new_host},
-            old_host: {"host": old_host},
-        }
-        hass.async_create_task.assert_called_once()
+from tests.common import MockConfigEntry, mock_coro
 
 
-async def test_async_setup_entry():
-    """Test setup of an entry."""
+async def test_empty_config(hass):
+    """Test with an empty config."""
+    assert await async_setup_component(hass, dynalite.DOMAIN, {}) is True
+    assert len(hass.config_entries.flow.async_progress()) == 0
+    assert hass.data[dynalite.DOMAIN] == {dynalite.DATA_CONFIGS: {}}
 
-    def async_mock(mock):
-        """Return the return value of a mock from async."""
 
-        async def async_func(*args, **kwargs):
-            return mock()
+async def test_async_setup(hass):
+    """Test a successful setup."""
+    host = "1.2.3.4"
+    port = 789
+    with patch.object(
```

Comment:
This should be enough:
```py
with patch("homeassistant.components.dynalite.DynaliteBridge.async_setup", return_value=True):
```

Note that the bridge setup function never returns anything else than `True` which seems to be a bug. We should probably return `False` if we fail to set up the bridge connection.

Reply:
the connection is happening asynchronously in the library, so will be quite difficult to test when we set it up. Ok if I defer this to later?

Reply:
We should at least clean up the code that pretends to check for a valid setup.

Reply:
fixed